src/main.rs

```diff
@@ -227,7 +214,7 @@ fn show_help() {
     println!("  -h | --help  : Display this help");
 }
 
-fn main() {
+fn main() -> Result<()> {
```

Comment:
You never return an error here, so is this change useful? ;)

Reply:
This is what the whole thing boils down to, because of `?` in `if !run_check(&arg, &gir_file)? {` :)

Reply:
Oh damn you're right! I completely missed it...